File: src/main/java/org/goafabric/calleservice/Application.java
package org.goafabric.calleservice;

import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.actuate.autoconfigure.security.servlet.ManagementWebSecurityAutoConfiguration;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration;
import org.springframework.boot.autoconfigure.security.servlet.UserDetailsServiceAutoConfiguration;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.Bean;

/**
 * Created by amautsch on 26.06.2015.
 */

@Slf4j
@SpringBootApplication(proxyBeanMethods = false,
        exclude = { SecurityAutoConfiguration.class, ManagementWebSecurityAutoConfiguration.class, UserDetailsServiceAutoConfiguration.class})
public class Application {

    public static void main(String[] args){
        SpringApplication.run(Application.class, args);
    }

    @Bean
    public CommandLineRunner init(ApplicationContext context) {
        return args -> {
            if ((args.length > 0) && ("-check-integrity".equals(args[0]))) {
                log.info("##checking integrity ...");
                if (true) {
                    throw new IllegalStateException("yo");
                }
                SpringApplication.exit(context, () -> 0);
            }
        };
    }
}

File: src/main/kotlin/org/goafabric/calleeservice/Application.kt
package org.goafabric.calleeservice

import org.goafabric.calleeservice.controller.CalleeController
import org.springframework.ai.tool.ToolCallbackProvider
import org.springframework.ai.tool.method.MethodToolCallbackProvider
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean


@SpringBootApplication
class Application

fun main(args: Array<String>) {
    runApplication<Application>(*args)
}

@Bean
fun weatherTools(calleeController: CalleeController): ToolCallbackProvider {
    return MethodToolCallbackProvider.builder().toolObjects(calleeController).build()
}